src/test/java/selenium/patterns/LoginTestWithPageFactory.java

```diff
@@ -0,0 +1,41 @@
+package selenium.patterns;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import selenium.Driver;
+import selenium.TestUser;
+import selenium.pages.LoginPageWithPF;
+import selenium.pages.ProductPageWithPF;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class LoginTestWithPageFactory {
+    private static final String INVENTORY_URL = "https://www.saucedemo.com/inventory.html";
+    private LoginPageWithPF loginPage;
+    private ProductPageWithPF productPage;
```

Comment:
no need to use pages as fields of the class

Reply:
The same as above. More details are needed.